client/components/admin/UserViewPage.jsx

```diff
@@ -0,0 +1,93 @@
+import React, { PropTypes } from 'react';
+import { connect } from 'react-redux';
+import ListDetails from '../common/ListDetails.jsx';
+
+class UserViewPage extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      fullname: Object.assign({}, props.userValue).name,
+      username: Object.assign({}, props.userValue).username,
+      email: Object.assign({}, props.userValue).email,
+      roleId: Object.assign({}, props.userValue).roleId,
+      createdAt: Object.assign({}, props.userValue).createdAt,
+    };
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (this.props.userValue.id !== nextProps.userValue.id) {
+      // Necessary to populate form when existing documents loaded directly.
+      this.setState({
```

Comment:
Since you are using the exact same data and just passing it into your props, you can just go ahead to use your props like that. If you want to proceed with this (which I don't advice and is more difficult) you can use  `this.setState({...nextProps.userValue, fullname: nextProps.userValue.name})`.

I see in several places that you convert the `name` prop into `fullname` for use in your component. Chose one and let it be consistent throughout your code.